### PR TITLE
Fix/intro to anchor frontend

### PR DESCRIPTION
**Pull request by thearyanag**
### Problem

The content was made for older versions of anchor, which can confuse new developers who are trying with anchor 0.30.0 or greater. 
The Lab repo attached is also using `@project-serum/anchor": "^0.25.0"` which can be considered as obsolete.

### Summary of Changes

Replaced the `IDL` with a new IDL generated by anchor 0.30.0 
Added `IDL` types to show how to use them
Updated the Lab example to support types 
Made a PR to the labs repo to update the version and the contract by 
https://github.com/Unboxed-Software/anchor-counter-program/tree/main 

PR(s) to the Labs repo https://github.com/Unboxed-Software/anchor-ping-frontend/pull/5

### CODE REVIEW

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Thanks Aryan! This looks great - please use Anchor 0.30.1 for Anchor as that's the latest version, the changes are probably small though. I've also cloned the anchor app to https://github.com/solana-developers/anchor-ping-frontend - please update there!

**Comment by thearyanag** [comment]
Hey @mikemaccana , I've made a PR to the counter program repos to bump the version to 0.30.1. Please find the respective PRs 
1. https://github.com/Unboxed-Software/anchor-counter-program/pull/4
2. https://github.com/Unboxed-Software/anchor-counter-program/pull/5
3. https://github.com/Unboxed-Software/anchor-counter-program/pull/6

Thanks for adding me as a collaborator, I'll make the required changes and make a PR.

**Comment by thearyanag** [comment]
Hey @mikemaccana , can you please guide me on how to merge my changes. I created a new project using the `create-solana-dapp` and the git history is not the same for the https://github.com/solana-developers/anchor-ping-frontend and https://github.com/thearyanag/anchor-ping-frontend/tree/feat/create-solana-dapp ,therefore can't create a PR and also unable to merge without using the `--force` flag.

I'm attaching SS on how the app is looking right now, 
image

Here are a transactions, the images are hyperlinked to the explorer

[image](https://explorer.solana.com/tx/5KKAiMAJe44yZMqidrL4Q6RQ9ZRypMbUMVTaqXZRUwjkXbDmegzR2LXBEaps9NFSJ4Pbe775SyzVvgDAWiFVDM5s?cluster=devnet)

[image](https://explorer.solana.com/tx/4jaqcPRBxL7GWkUWra1b3sRDPUmWthHVWqnf1GsVrfnmW9AHC7nJrtsg5ZgiuWkDTEDttqYYN1Hwcm1auPu3homz?cluster=devnet)

File: content/courses/onchain-development/intro-to-anchor-frontend.md [hunk 1]
```diff
@@ -291,7 +420,14 @@ const provider = new AnchorProvider(connection, wallet, {});
 setProvider(provider);
 
 const programId = new PublicKey("JPLockxtkngHkaQT5AuRYow3HyUv5qWzmhwsCPd653n");
```
Comment (jereldlimjy) [inline content/courses/onchain-development/intro-to-anchor-frontend.md:422]:
```suggestion
```

**Review by jereldlimjy — CHANGES_REQUESTED** [review]
As of Anchor 0.30.1, the `Program` class no longer requires a `programId` in its constructor since the updated IDL stores the `programId` under the `address` field (see [here](https://coral-xyz.github.io/anchor/ts/classes/Program.html)).

Other mentions of the `Program` class or `programId` which I've not commented on should be amended accordingly (I am not able to add comments/suggestions on those in this PR).

**Comment by thearyanag** [comment]
> As of Anchor 0.30.1, the `Program` class no longer requires a `programId` in its constructor since the updated IDL stores the `programId` under the `address` field (see [here](https://coral-xyz.github.io/anchor/ts/classes/Program.html)).
> 
> Other mentions of the `Program` class or `programId` which I've not commented on should be amended accordingly (I am not able to add comments/suggestions on those in this PR).

Hey @jereldlimjy , appreciate you doing this, i was still in middle of updating the whole document to be in-synch with the anchor repo.

**Comment by thearyanag** [comment]
Hey @mikemaccana , I have 
- made the changes in the content (including the images for logs and gif for the demo) , 
- used `create-solana-dapp` to scaffold the app
- made the required changes for `starter` ,`solution-increment` and ``solution-decrement` branch, which can be seen at https://github.com/thearyanag/anchor-ping-frontend

I'll create a PR to the https://github.com/solana-developers/anchor-ping-frontend , when the [PR](https://github.com/solana-developers/anchor-ping-frontend/pull/1) to avoid any merge issues.

I've also attached the demo gift, the quality decreased while converting from mp4 to gift. 

anchor-frontend-demo

**Comment by jereldlimjy** [comment]
> > As of Anchor 0.30.1, the `Program` class no longer requires a `programId` in its constructor since the updated IDL stores the `programId` under the `address` field (see [here](https://coral-xyz.github.io/anchor/ts/classes/Program.html)).
> > Other mentions of the `Program` class or `programId` which I've not commented on should be amended accordingly (I am not able to add comments/suggestions on those in this PR).
> 
> Hey @jereldlimjy , appreciate you doing this, i was still in middle of updating the whole document to be in-synch with the anchor repo.

No problem! Let me know if you need any help, I can PR directly to your branch if need be :)

File: content/courses/onchain-development/intro-to-anchor-frontend.md [hunk 2]
```diff
@@ -271,16 +394,17 @@ and listen to events. You can also
...
 explicitly specified.
 
+To enable type support, import the types to your project from `/target/types`
+present in your anchor project, and declare the type for the program object.
+
 All together, the final setup looks something like this:
 
 ```typescript
 import idl from "./idl.json";
+import type { CounterContract } from "@/types";
```
Comment (mikemaccana) [inline content/courses/onchain-development/intro-to-anchor-frontend.md:407]: Call it a program not a contract. This is mentioned in CONTRIBUTING, also https://solana.com/docs/terminology#onchain-program

**Review by mikemaccana — APPROVED** [review]
I'm happy to mark this as the winner, can you please fix the minor issue above and let me know if there are any outstanding PRs that need to be merged?

**Comment by thearyanag** [comment]
Hey @mikemaccana , I've fixed the issue; these are the final PRs left to be merged. 

https://github.com/solana-developers/anchor-ping-frontend/pull/4
https://github.com/solana-developers/anchor-ping-frontend/pull/3
https://github.com/solana-developers/anchor-ping-frontend/pull/2

Thanks for your help throughout! Please do let me know, if other lessons need attention.

**Pull request merged**